### ISSUE TITLE
Can't find annotation_01362.pk

### ISSUE THREAD

**Issue by haoranD** [zfchenUnique/DCL-Release#3]
Hi,

Thanks for your amazing work.

I am trying to run the code, but I got a problem.
The program can NOT find the '/tubeProposalsRelease/1.0_1.0_0.6_0.7/annotation_01362.pk'
I also checked the code, and I can NOT find the code for generating the '/tubeProposalsRelease/1.0_1.0_0.6_0.7/annotation_01362.pk'.

Could you please give me any suggestions? Thanks for your help.

**Comment by zfchenUnique** [comment]
Hi Haoran, 
      
      Thanks for your interest in the work.
      You need to connect the mask-rcnn proposals to generate the object trajectories first and then start training DCL.
      We provide the connected tubes of test set (https://drive.google.com/file/d/18NkK29Hnu3hMPaenveB5g4Lp9WRdiIA7/view) in  **fast evaluation**. For training, you need to download image proposals (http://data.csail.mit.edu/clevrer/derender_proposals.zip) and connect them into tube proposals as described in **Step-by-step Training**.
       I hope this can resolve your concerns!

Regards,
Zhenfang

**Comment by haoranD** [comment]
Really appreciated your reply.

Yes, I am following the **Step-by-step Training**
I downloaded the proposals, then when I am running the commend you gave in Step 1:
```
python scripts/script_gen_tube_proposals.py \
    --start_index 10000 \
    --end_index 15000 \
    --attr_w 0.6 \
    --match_thre 0.7 \
    --version 2 \
    --visualize_flag 0 \
    --tube_folder_path ../clevrer/tubeProposalsRelease \
    --use_attr_flag 0 \
```
It indeed generated the proposal_*.pk files in tubeProposalsRelease, but there is no annotation_*.pk generated in any folders.
I am sorry that I may miss something, but could you please give me any suggestions?

Best Wishes.

**Comment by zfchenUnique** [comment]
Hi,
   Sorry for misunderstanding what you mean.
   When generating CLEVRER data, the original authors do not keep the object bounding annotations. Thus, I generate "pseudo ground truth" by matching attributes provided in the proposals to generate "annotation_.pk".
   1. You may generate your "annotation_.pk" by using attributes in http://data.csail.mit.edu/clevrer/derender_proposals.zip.
   2. Alternatively, you may just skip and comment evaluation function **compute_recall_and_precision(opt)** in script_gen_tube_proposals.py. There is no need to generate the GT for running DCL. It is just a measure to evaluate the recall and precision of the generated proposals. You may visualize the generate tube proposals to check the proposal quality.
   
Regards,
Zhenfang

**Comment by haoranD** [comment]
Hi,

Thanks for your time and fast response. I will first try the second solution since it is the simplest one.

Sorry, may I ask if there is any code I can refer to for the first solution you recommended to generate the "annotation_.pk"?
Because I saw the "annotation_.pk" is also quite useful in later training stage (e.g., trainval_tube_v2.py)

Best wishes,
Haoran

**Comment by zfchenUnique** [comment]
HI,
   To generate "annotation_.pk", you just need to simply align objects by their three visible attributes (color, shape, and material). As described in CLEVRER, each object can be uniquely referred to by these three visible attributes.

**Comment by haoranD** [comment]
Many thanks for your help.

I don't have any questions, and I closed the issue for now.

Have a good one!

**Comment by Natithan** [comment]
Hi,

For the lazy: do you happen to have the tubes for the train and validation set laying around, so I don't need to generate them myself? :)
